v2/dtos/requests/provisionwatcher_test.go

```diff
@@ -0,0 +1,328 @@
+			assert.Equal(t, tt.expectError, err != nil, "Unexpected addDeviceRequest validation result.", err)
+		})
+	}
+}
+
+func TestAddProvisionWatcherRequest_UnmarshalJSON(t *testing.T) {
+	valid := testAddProvisionWatcher
+	resultTestBytes, _ := json.Marshal(testAddProvisionWatcher)
+	type args struct {
+		data []byte
+	}
+	tests := []struct {
+		name                string
+		addProvisionWatcher AddProvisionWatcherRequest
+		args                args
+		wantErr             bool
+	}{
+		{"unmarshal AddProvisionWatcherRequest with success", valid, args{resultTestBytes}, false},
+		{"unmarshal invalid AddProvisionWatcherRequest, empty data", AddProvisionWatcherRequest{}, args{[]byte{}}, true},
+		{"unmarshal invalid AddProvisionWatcherRequest, string data", AddProvisionWatcherRequest{}, args{[]byte("Invalid AddDeviceRequest")}, true},
+	}
+	fmt.Println(string(resultTestBytes))
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			var expected = tt.addProvisionWatcher
+			err := tt.addProvisionWatcher.UnmarshalJSON(tt.args.data)
+			if tt.wantErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+				assert.Equal(t, expected, tt.addProvisionWatcher, "Unmarshal did not result in expected AddProvisionWatcherRequest.")
+			}
+		})
+	}
+}
+
+func TestAddProvisionWatcherReqToProvisionWatcherModels(t *testing.T) {
+	requests := []AddProvisionWatcherRequest{testAddProvisionWatcher}
+	expectedProvisionWatcherModel := []models.ProvisionWatcher{
+		{
+			Name:   testProvisionWatcherName,
+			Labels: testProvisionWatcherLabels,
+			Identifiers: map[string]string{
+				"address": "localhost",
+				"port":    "3[0-9]{2}",
+			},
+			BlockingIdentifiers: map[string][]string{
+				"port": {"397", "398", "399"},
+			},
+			ServiceName: TestDeviceServiceName,
+			ProfileName: TestDeviceProfileName,
+			AdminState:  models.Locked,
+			AutoEvents: []models.AutoEvent{
+				{Resource: "TestDevice", Frequency: "300ms", OnChange: true},
+			},
+		},
+	}
+	resultModels := AddProvisionWatcherReqToProvisionWatcherModels(requests)
+	assert.Equal(t, expectedProvisionWatcherModel, resultModels, "AddProvisionWatcherReqToProvisionWatcherModels did not result in expected ProvisionWatcher model.")
+}
+
+func TestUpdateProvisionWatcherRequest_Validate(t *testing.T) {
+	emptyString := " "
+	emptyMap := make(map[string]string)
+	invalidUUID := "invalidUUID"
+
+	valid := testUpdateProvisionWatcher
+	noReqId := valid
+	noReqId.RequestId = ""
+	invalidReqId := valid
+	invalidReqId.RequestId = invalidUUID
+
+	// id
+	validOnlyId := valid
+	validOnlyId.ProvisionWatcher.Name = nil
+	invalidId := valid
+	invalidId.ProvisionWatcher.Id = &invalidUUID
+	// name
+	validOnlyName := valid
+	validOnlyName.ProvisionWatcher.Id = nil
+	invalidEmptyName := valid
+	invalidEmptyName.ProvisionWatcher.Name = &emptyString
+	// no id and name
+	noIdAndName := valid
+	noIdAndName.ProvisionWatcher.Id = nil
+	noIdAndName.ProvisionWatcher.Name = nil
+
```

Comment:
Add one for invalid character(s) in name. i.e. edgex-dto-rfc3986-unreserved-chars fails.